src/golang/lib/workflow/artifact/artifact.go

```diff
@@ -222,11 +221,18 @@ func (a *ArtifactImpl) updateArtifactResultAfterComputation(
 		)
 		if err != nil {
 			log.Errorf("Unable to read artifact result metadata from storage and unmarshal: %v", err)
-			return
+			execState.Error.Context = fmt.Sprintf(
+				"%s\nError reading metadata: %v",
+				execState.Error.Context,
```

Comment:
is `execState.Error != nil` in this case? I worry this will panic if there wasn't an issue with the computation, but something bad happened when we ReadFromStorage.

Reply:
yeah that's a good catch, we should check the pointer state before assigning values